src/main/java/com/artipie/asto/google/GoogleStorage.java

```diff
@@ -107,15 +146,16 @@ public GoogleStorage(final Vertx vertx, final WebClient client, final String buc
     public CompletableFuture<Content> value(final Key key) {
         final ReactiveWriteStream<Buffer> stream = ReactiveWriteStream
             .writeStream(this.vertx.getDelegate());
-        this.client.get(String.format(GoogleStorage.GET_URL, this.bucket, key.string()))
+        this.client.get(
+            this.port,
+            this.host,
+            String.format(GoogleStorage.GET_REQUEST, this.bucket, key.string())
+        )
             .as(BodyCodec.pipe(WriteStream.newInstance(stream)))
             .rxSend().subscribe();
-        return  Flowable.fromPublisher(stream).flatMapCompletable(
-            buffer -> Completable.fromSingle(
-                Single.just(new Content.From(buffer.getBytes()))
-            ))
-            .to(CompletableInterop.await())
-            .thenApply(content -> (Content) content).toCompletableFuture();
+        final Flowable<ByteBuffer> flow = Flowable.fromPublisher(stream)
+            .map(buffer -> ByteBuffer.wrap(buffer.getBytes()));
+        return CompletableFuture.supplyAsync(() -> new Content.From(flow));
```

Comment:
@Guseyn there is no need to wrap the constructor with `supplyAsync` - it switches the context just to create new object (without any logic in constructor). I think you can move `this.client.get(...)` call to this `supplyAsync()` or use  `rxSend().map(rsp -> new Content.From(flow))` and convert it to future using single interop.

Reply:
@g4s8 I create Flowable from the `stream` that we used to write result into, not for `this.client.get(...)`, that's why `rxSend().map` will map `Function<? super HttpResponse<Void>, ?>`, not `Function<? super Buffer, ? extends Buffer>`, where `Buffer` is **Vertx** buffer, which I have to wrap with `ByteBuffer`.

Reply:
@g4s8 I guess I understand now what you mean. okay

Reply:
@Guseyn I mean you're creating a new thread just to instantiate an object, please don't use `supplyAsync` to create new object. There are two possible ways to fix it:
 - use `CompletableFuture.completedFuture(new Content.From(flow))`
 - use RxJava to map single to `CompletableFuture`

Reply:
@g4s8 Thanks, it fixed.